backend/src/main/kotlin/app/ehrenamtskarte/backend/userdata/database/Schema.kt

```diff
@@ -0,0 +1,33 @@
+package app.ehrenamtskarte.backend.userdata.database
+
+import app.ehrenamtskarte.backend.projects.database.Projects
+import org.jetbrains.exposed.dao.IntEntity
+import org.jetbrains.exposed.dao.IntEntityClass
+import org.jetbrains.exposed.dao.id.EntityID
+import org.jetbrains.exposed.dao.id.IntIdTable
+import org.jetbrains.exposed.sql.javatime.CurrentTimestamp
+import org.jetbrains.exposed.sql.javatime.timestamp
+
+object UserEntitlements : IntIdTable() {
+    val userHash = binary("userHash")
+    val startDate = timestamp("startDate")
+    val endDate = timestamp("endDate")
+    val revoked = bool("revoked")
+    val projectId = reference("projectId", Projects)
```

Comment:
We should probably have a `regionId` rather than a project id here as cards are always assigned to a region rather than a project

Reply:
I think this should be disscussed then.
Because the current idea was that we extract the projectId based on the authorization token of the user who performs the import.
Do we want to have a region-level token?
Or should be query regionId by projectId?
What if the project has more than 1 region?

Reply:
We have a user based token and the user belongs to a region and to a project. so thats fine.
If the project has more region the user that imports for what region the cards can be created.

Reply:
So the idea is to retrieve the region from the "user" making the "PUT API request" and then storing this region in the table `UserEntitlement`?

Reply:
Now i understand the issue from victoria.
Lets say we use the import for a project like bavaria and we have one project manager or a similar role that doesn't apply to a region because it wants to import users/cards from different regions. Then in the import csv should be a field `regionId` and we just check to which `regionId` belongs the project and check if the api token of the user is valid for this project.
The api token is user based (https://github.com/digitalfabrik/entitlementcard/issues/1491) so it belongs to a project

Nevertheless we need a regionId to create a card and we should use it as a reference here.
To keep this as generic as possible we should probably send a `regionId/regionIdentifier` in the csv and if its valid set the reference

Reply:
So, I make regionId column instead of projectId and fill it from csv.
And also add a validation, that the passed regionId belongs to the project, associated with the user.
Right?

Reply:
So, I make regionId column instead of projectId and fill it from csv.
And also add a validation, that the passed regionId belongs to the project, associated with the user.